ui/components/component-library/banner-base/banner-base.types.ts

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+
+import type {
+  PolymorphicComponentPropWithRef,
+  StyleUtilityProps,
+} from '../box';
+
+import type { TextProps } from '../text';
+import type { ButtonLinkProps } from '../button-link';
+import type { ButtonIconProps } from '../button-icon';
+
+/**
+ * Makes all props optional so that if a prop object is used not ALL required props need to be passed
+ * TODO: Move to appropriate place in app as this will be highly reusable
+ */
+type MakePropsOptional<T> = {
+  [K in keyof T]?: T[K];
+};
```

Comment:
This type helper allows for us to use prop objects without needing to pass all required props through the prop object.


https://github.com/MetaMask/metamask-extension/assets/8112138/74ec6ee8-998c-4621-b64b-acb9d00884fd

Reply:
cc @garrettbear